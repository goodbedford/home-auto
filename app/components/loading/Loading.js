import React from "react";

const PropTypes = React.PropTypes;

const style = {
  alignItems: "center",
  background: "linear-gradient(moccasin	,Bisque,pink)",
  // background: "linear-gradient(moccasin	,orange,pink)",
  color: "SteelBlue",
  display: "flex",
  fontSize: "36px",
  fontWeight: "bold",
  height: "50vh",
  justifyContent: "center",
  width: "100%"
};

const propTypes = {
  loadingMsg: PropTypes.string.isRequired,
  styles: PropTypes.string.isRequired
};


const defaultProps = {
  loadingMsg: "Loading",
  styles: "loading loading--block-md"
};

class Loading extends React.Component {
  constructor(props) {
    super(props);
    this.msg = this.props.loadingMsg;
    this.state = {
      msg: this.msg,
      isMounted: false
    };
  }
  componentDidMount() {
    // debugger
    this.setState({isMounted:true});

    if(this.state.isMounted) {
      this.intervalId = setInterval(() => {
        this.setState((prevState, props) => {
          if(prevState.msg.length === 13) {
            return {msg: this.msg};
          }
          return {msg: prevState.msg + "."};
        });
      },500);
    }
  }
  componentWillUnMount() {
    this.setState({isMounted:false});
    console.log("clear Loading",this);
    clearInterval(this.intervalId);

  }

  render() {
    return (
      <div className={this.props.styles}>{this.state.msg}</div>
    );
  }
}
Loading.propTypes = propTypes;
Loading.defaultProps = defaultProps;
export default Loading;
