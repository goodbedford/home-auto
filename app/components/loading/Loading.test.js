/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Loading from "./Loading";

describe("<Loading />", () => {
  it("Loading snapshot", () => {
    const wrapper = shallow(<Loading />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should have correct initial state", () => {
    const wrapper = shallow(<Loading />);

    const msg = wrapper.state("msg");
    const isMounted = wrapper.state("isMounted");

    expect(isMounted).toEqual(false);
    expect(msg).toBe("loading");
  });
  it("should have correct props", () => {
    const wrapper = shallow(<Loading />);

    const text = wrapper.props().children;
    const styles = wrapper.props().className;

    expect(text).toBe("loading");
    expect(styles).toBe("loading loading--block-md");
  });

});
