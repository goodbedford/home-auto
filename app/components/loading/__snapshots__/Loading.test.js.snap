// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Loading snapshot 1`] = `
<div
  className="loading loading--block-md"
>
  loading
</div>
`;
